vllm/attention/backends/flash_attn.py

```diff
@@ -105,6 +105,11 @@ class FlashAttentionMetadata(AttentionMetadata):
     # seq_lens stored as a tensor.
     seq_lens_tensor: Optional[torch.Tensor]
 
+    # (batch_size,). The orig sequence length per sequence.
+    orig_seq_lens: Optional[List[int]]
+    # orig_seq_lens stored as a tensor.
+    orig_seq_lens_tensor: Optional[torch.Tensor]
```

Comment:
IMO this PR shouldn't need to touch the `flash_attn` backend. Could you say a bit about why you added these?

Reply:
I added the orig_seq_lens to the flash-attn backend metadata then dual-chunk-flash-attn can inherit most of the logic of metadata building from flash-attn to avoid code duplication.